File: apps/api/app/schemas/report.py
"""Report schemas."""
from typing import Optional
from pydantic import BaseModel


class ReportGenerateRequest(BaseModel):
    auditId: str
    format: str = "markdown"


class ReportResponse(BaseModel):
    id: str
    auditId: str
    userId: str
    contractId: str
    format: str
    title: str
    summary: str
    hash: str
    ipfsHash: Optional[str] = None
    ipfsUri: Optional[str] = None
    txHash: Optional[str] = None
    generatedBy: str
    createdAt: str


class ReportExportRequest(BaseModel):
    format: str = "pdf"

File: packages/shared/src/types/report.ts
export interface Report {
  id: string;
  auditId: string;
  userId: string;
  contractId: string;
  format: ReportFormat;
  title: string;
  content: string;
  summary: string;
  hash: string;
  ipfsHash: string | null;
  ipfsUri: string | null;
  txHash: string | null;
  generatedBy: string;
  metadata: ReportMetadata;
  createdAt: string;
}

export enum ReportFormat {
  MARKDOWN = "markdown",
  JSON = "json",
  PDF = "pdf",
  HTML = "html",
}

export enum ReportSection {
  EXECUTIVE_SUMMARY = "executive_summary",
  VULNERABILITY_DETAILS = "vulnerability_details",
  GAS_ANALYSIS = "gas_analysis",
  RECOMMENDATIONS = "recommendations",
  CODE_COVERAGE = "code_coverage",
  APPENDIX = "appendix",
  AI_INSIGHTS = "ai_insights",
  AUDIT_METADATA = "audit_metadata",
}

export interface ReportMetadata {
  generator: string;
  version: string;
  auditType: string;
  contractName: string;
  contractAddress: string | null;
  securityScore: number;
  totalVulnerabilities: number;
  sections: ReportSection[];
  aiAssisted: boolean;
  onChainStored: boolean;
}

export interface ReportTemplate {
  id: string;
  name: string;
  description: string;
  format: ReportFormat;
  sections: ReportSection[];
  styling: Record<string, unknown>;
  customCSS: string | null;
}
